File: words/reverse.js
// JavScript implementation to reverse a string
// using recursion
function reverseString(str) {
    if (str === "") {
    return "";
    } else {
    return reverseString(str.substr(1)) + str.charAt(0);
    }
   }
   str1 = "MUO";
   str2 = "Welcome to MUO";
   str3 = "She sells seashells by the seashore";
   document.write("Input string: <br>");
   document.write(str1 + "<br>");
   document.write(str2 + "<br>");
   document.write(str3 + "<br>");
   str1 = reverseString(str1);
   str2 = reverseString(str2);
   str3 = reverseString(str3);
   document.write("Reversed string: <br>");
   document.write(str1 + "<br>");
   document.write(str2 + "<br>");
   document.write(str3 + "<br>");

File: words/reverse.py


def reverseString1(str):
    """
    Python implementation to reverse a string
    using extended slice syntax
    """
    return str[::-1]


def reverseString2(str):
    """
    Python implementation to reverse a string
    using recursion
    """
    if len(str) == 0:
        return str
    else:
        return reverseString2(str[1:]) + str[0]


def reverseString3(str):
    """
    Python implementation to reverse a string
    using reversed method()
    """
    str = "".join(reversed(str))
    return str


def reverseString4(str):
    """
    Python implementation to reverse a string
    using a temporary string
    """
    tempStr = ""
    for s in str:
        tempStr = s + tempStr
    return tempStr


# Variables
def vars():
    str = [
        "MUO",
        "Welcome to MUO",
        "She sells seashells by the seashore"
    ]
    return str


# TESTS
def tst1():
    str = vars()
    print("Input string:")
    print(str[0])
    print(str[1])
    print(str[2])
    str1 = reverseString1(str[0])
    str2 = reverseString1(str[1])
    str3 = reverseString1(str[2])
    print("Reversed string:")
    print(str1)
    print(str2)
    print(str3)


def tst2():
    str = vars()
    print("Input string:")
    print(str[0])
    print(str[1])
    print(str[2])
    str1 = reverseString2(str[0])
    str2 = reverseString2(str[1])
    str3 = reverseString2(str[2])
    print("Reversed string:")
    print(str1)
    print(str2)
    print(str3)


def tst3():
    str = vars()
    print("Input string:")
    print(str[0])
    print(str[1])
    print(str[2])
    str1 = reverseString3(str[0])
    str2 = reverseString3(str[1])
    str3 = reverseString3(str[2])
    print("Reversed string:")
    print(str1)
    print(str2)
    print(str3)


def tst4():
    str = vars()
    print("Input string:")
    print(str[0])
    print(str[1])
    print(str[2])
    str1 = reverseString4(str[0])
    str2 = reverseString4(str[1])
    str3 = reverseString4(str[2])
    print("Reversed string:")
    print(str1)
    print(str2)
    print(str3)


# RUN TESTS
tst4()
